src/nvapi_gpu.cpp

```diff
@@ -469,13 +469,50 @@ extern "C" {
         if (pGpuArchInfo->version != NV_GPU_ARCH_INFO_VER_1 && pGpuArchInfo->version != NV_GPU_ARCH_INFO_VER_2)
             return IncompatibleStructVersion(n);
 
-        auto architectureId = adapter->GetArchitectureId();
+        NV_GPU_ARCHITECTURE_ID architectureId = {};
+
+        if (auto archOverride = env::getEnvVariable("DXVK_NVAPI_GPU_ARCH"); !archOverride.empty()) {
+#define CHECK_ARCH(arch)       \
+    if (archOverride == #arch) \
+        architectureId = NV_GPU_ARCHITECTURE_##arch;
+            CHECK_ARCH(T2X)
```

Comment:
> I think we should only handle those that return Ok.

Sure, that makes sense.

> Since that list would be limited, I would also be fine to allow terms like Ada, Ampere etc. But AD, GA or AD100, GA100 looks fine too.

That would require some fancy case handling which I wanted to avoid.

> Would be cool though if the env var value would be case insensitive.

If only that wasn't so unreasonably hard in C++ without something like ICU :weary:

I guess I can do `toupper` on the env var value but I admit user friendlyness was not the goal here. I can add a warning that the override string was not recognized though, so at least it's obvious that override didn't do anything.

Reply:
> I guess I can do toupper on the env var value but I admit user friendlyness was not the goal here. I can add a warning that the override string was not recognized though, so at least it's obvious that override didn't do anything.

I had only `toupper` in my mind. But perfectly cool to keep it simple and strict. So let's indeed go with the values like you suggested (without alternative values) combined with log statements for invalid values and successful overrides by anv var.